File: Iniciante/1045.rs
use std::io;

fn main() {
  let mut input = String::new();

  io::stdin().read_line(&mut input).unwrap();

  let mut values: Vec<f64> = input.split_whitespace()
    .map(|x| x.parse().unwrap()) 
    .collect(); 

  values.sort_by(|a, b| b.partial_cmp(a).unwrap()); 

  let a = values[0];
  let b = values[1];
  let c = values[2];

  if a >= b + c {
    println!("NAO FORMA TRIANGULO");
    return
  }

  let pow_a = a * a;
  let pow_b = b * b;
  let pow_c = c * c;

  if pow_a == pow_b + pow_c {
    println!("TRIANGULO RETANGULO");
  } else if pow_a > pow_b + pow_c {
    println!("TRIANGULO OBTUSANGULO");
  } else {
    println!("TRIANGULO ACUTANGULO");
  }

  if a == b && b == c {
    println!("TRIANGULO EQUILATERO");
  } else if a == b || b == c || a == c {
    println!("TRIANGULO ISOSCELES");
  }
}

File: Iniciante/1045/1045.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Collections;

public class Main {
  public static void main(String[] args) throws IOException {
    InputStreamReader stream = new InputStreamReader(System.in);
    BufferedReader reader = new BufferedReader(stream);

    String[] values = reader.readLine().split(" ");

    Double[] parsed = {
        Double.parseDouble(values[0]),
        Double.parseDouble(values[1]),
        Double.parseDouble(values[2])
    };
    Collections.sort(Arrays.asList(parsed), Collections.reverseOrder());

    double a = parsed[0];
    double b = parsed[1];
    double c = parsed[2];

    if (a >= b + c) {
      System.out.println("NAO FORMA TRIANGULO");
      return;
    }

    double pow_a = a * a;
    double pow_b = b * b;
    double pow_c = c * c;

    if (pow_a == pow_b + pow_c) {
      System.out.println("TRIANGULO RETANGULO");
    } else if (pow_a > pow_b + pow_c) {
      System.out.println("TRIANGULO OBTUSANGULO");
    } else if (pow_a < pow_b + pow_c) {
      System.out.println("TRIANGULO ACUTANGULO");
    }

    if (a == b && b == c) {
      System.out.println("TRIANGULO EQUILATERO");
    } else if (a == b || b == c || c == a) {
      System.out.println("TRIANGULO ISOSCELES");
    }
  }
}
